modules/nf-core/nanocomp/main.nf

```diff
@@ -0,0 +1,138 @@
+    tuple val(meta), path("*NanoStats.txt"), emit: stats_txt
+    path "versions.yml", emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: ""
+    print(prefix)
+    if (prefix == ""){
+        prefixflag = ""
+    } else {
+        prefixflag = "--prefix " + prefix
+    }
+
+    //determine input file type
+    filetypes = []
+    for (file in filelist){
+        tokenized_filename = file.getName().tokenize('.')
+        if (tokenized_filename.size() < 2){
+            throw new java.lang.IndexOutOfBoundsException("Every input file to nanocomp has to have a file ending.")
+        }
+
+        first_namepart = true
+        extension_found = false
+
+        for (namepart in tokenized_filename){
+            if (namepart == ""){
+                continue
+            }
+
+            // prevent the file name to be seen as extension
+            if (first_namepart == true){
+                first_namepart = false
+                continue
+            }
+
+            if (["fq","fastq"].contains(namepart)){
+                filetypes.add("fastq")
+                extension_found = true
+                break
+            } else if (["fasta", "fna", "ffn", "faa", "frn", "fa"].contains(namepart)) {
+                filetypes.add("fasta")
+                extension_found = true
+                break
+            } else if (namepart == "bam") {
+                filetypes.add("bam")
+                extension_found = true
+                break
+            } else if (namepart == "txt") {
+                filetypes.add("summary")
+                extension_found = true
+                break
+            }
+        }
+
+        if (extension_found == false){
+            throw new java.lang.IllegalArgumentException("There was no suitable filetype found for " + file.getName() +
+            ". NanoComp only accepts fasta (fasta, fna, ffn, faa, frn, fa), fastq (fastq, fq), bam and Nanopore sequencing summary (txt).")
+        }
+    }
+
+    filetypes.unique()
+    if (filetypes.size() < 1){
+        throw new java.lang.IllegalArgumentException("There was no suitable filetype found in NanoComp input. Please use fasta, fastq, bam or Nanopore sequencing summary.")
+    }
+    if (filetypes.size() > 1){
+        throw new java.lang.IllegalArgumentException("You gave different filetypes to NanoComp. Please use only *one* of fasta, fastq, bam or Nanopore sequencing summary.")
+    }
+    filetype = filetypes[0]
+
+    """
+    NanoComp \\
+        --$filetype $filelist \\
+        --threads $task.cpus \\
+        $prefixflag
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        nanocomp: \$(echo \$(NanoComp --version 2>&1) | sed 's/^.*NanoComp //; s/Using.*\$//' ))
+    END_VERSIONS
+    """
+
+    stub:
```

Comment:
I honestly don't know what you mean by that, because the little `def prefix = ...` line is set also in the stub, additionally I added the prefix to every file like it would be the behavior in the nanocomp execution, i.e. `touch "${prefix}"NanoComp_lengths_violin.html`, so if that is not enough, can you elaborate what is missing? :)

Reply:
No worries :) Yes, with prefix "test_" the filenames are like "test_NanoComp_N50.html" :)

Reply:
Allright! This is perfect then, good job!

Reply:
thx!